fem/datacollection.cpp

```diff
@@ -1136,9 +1138,16 @@ void ParaViewDataCollection::SetCompression(bool compression_)
    // If we are enabling compression, and it was disabled previously, use the
    // default compression level. Otherwise, leave the compression level
    // unchanged.
-   if (compression_ && compression == 0)
+   // FIXME: the logic here does not exactly correspond to the doxygen
+   // documentation which states that previously set compression level restored
+   // if compression is disabled and then re-enabled.
+   if (compression_)
+   {
+      if (compression == 0) { SetCompressionLevel(-1); }
+   }
+   else
    {
-      SetCompressionLevel(-1);
+      SetCompressionLevel(0);
    }
```

Comment:
@pazner, I fixed the logic here to make sure the method actually disables compression when `compression_` is `false`. However, I think there is still a discrepancy between the doxygen documentation and the implementation, so I left the above FIXME comment. If you agree that this needs fixing, we can do it here, or in a separate PR. What do you think?

Reply:
Looks good to me, thanks for the fix!

Reply:
I'll take a stab at making the logic consistent with the comment in this PR.

Reply:
Does 01fde3cecb28349a6d6af1a97e1fb49964bb48fe look OK?

Reply:
It looks like `ParaViewDataCollection::SetCompression(true)` will not enable compression if `compression_level == 0` but maybe that's okay -- this is what the doxygen implies, kind of.

Reply:
Yeah, I agree it's a little strange, but it does somehow make sense… if we want different behavior, a special case could be if `compression_level` is zero, and the user calls `SetCompression(true)` then the compression level gets reset to the default.

It's also probably a corner case that nobody is likely to run into…